simple_app/app_simple.py

```diff
@@ -536,6 +541,37 @@ def camdplot():
     return plot
 
 
+@app_simple.route('/specplot')
+def specplot():
+    """
+    Creates the bokeh representation of the plot
+
+    Returns
+    -------
+    plot
+        jsonified bokeh plot for the spectra
+    """
+    db = SimpleDB(db_file, connection_arguments={'check_same_thread': False})  # open database
+    query: str = curdoc().template_variables['query']  # get the query (on page)
+    resultdict: dict = db.inventory(query)  # get everything about that object
+    everything = Inventory(resultdict)  # parsing the inventory
+    dfspecinfo: pd.DataFrame = everything.listconcat('Spectra', False)
+    print(dfspecinfo)
+    tooltips = []  # to appear on hover
+    p = figure(title=query, plot_width=800, plot_height=400,
+               active_scroll='wheel_zoom', active_drag='box_zoom',
+               tools='pan,wheel_zoom,box_zoom,hover,tap,reset', tooltips=tooltips,
+               sizing_mode='stretch_width')
+    tspec: Table = db.query(db.Spectra.c.spectrum).\
+        filter(db.Spectra.c.source == query).\
+        table(spectra=['spectrum'])
+    for i, spec in enumerate(tspec):
+        spec: Spectrum1D = spec[0]
+        p.line(x=spec.spectral_axis.value, y=spec.flux.value)
```

Comment:
If I understand this correctly: if we have multiple spectra, all of them would be plotted together in the same plot? I like that better than an individual plot per spectrum. However, we do need to be able to distinguish one spectrum from another so we should figure out how to add a label to the plot with perhaps the filename.

By adding unique labels we can also rely on the Bokeh legend to distinguish them and disable plotting one or another.

Reply:
Maybe there could be checkboxes for all the available spectra? So I could turn on/off any particular spectrum?

Reply:
100% to both of these; the main issue is to get interactivity I think I need to put everything in a ```ColumnDataSource``` object which has to be constructed from any homogeneous data source. Any recommended solutions to that would be much obliged! I could either make a separate CDS for each spectra (although I don't know how many there will be per query, and I need to pass them out of the function) or do some weird null padding in a dictionary to ensure homogeneity.